azurerm/config.go

```diff
@@ -59,11 +59,21 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure"
+	uuid "github.com/hashicorp/go-uuid"
 	"github.com/hashicorp/terraform/terraform"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/authentication"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
 )
 
+var msClientRequestID = ""
+
+func init() {
+	// Initialize UUID to pass through `x-ms-client-request-id` header.
+	msClientRequestID, _ = uuid.GenerateUUID()
```

Comment:
rather than having this in the init (and thus not being able to handle the error), I believe this'd be better within the `getArmClient` function in `./azurerm/config.go`? in which case this could become `if msClientRequestID = "" {/*generate*/}`

Reply:
is there any reason we couldn't use the Request ID returned from the Authorization API, rather than generating our own?

Reply:
re: error handling, i don't think we should honestly, its not worth crashing the run to track this.

Reply:
> is there any reason we couldn't use the Request ID returned from the Authorization API, rather than generating our own?

We could if that Request ID is uniquely generated each time when running `terraform plan` or `terraform apply`. Which Authorization API ? If it's from ARM API, it also needs to be correlated with other APIs within same session.

Reply:
> re: error handling, i don't think we should honestly, its not worth crashing the run to track this.

Agree.